packages/task/src/browser/task-configurations.ts

```diff
@@ -129,16 +129,14 @@ export class TaskConfigurations implements Disposable {
      *
      * The invalid task configs are not returned.
      */
-    async getTasks(token: number): Promise<TaskConfiguration[]> {
+    async getTasks(): Promise<TaskConfiguration[]> {
         const configuredTasks = Array.from(this.tasksMap.values()).reduce((acc, labelConfigMap) => acc.concat(Array.from(labelConfigMap.values())), [] as TaskConfiguration[]);
         const detectedTasksAsConfigured: TaskConfiguration[] = [];
-        for (const [rootFolder, customizations] of Array.from(this.taskCustomizationMap.entries())) {
+        for (const [, customizations] of Array.from(this.taskCustomizationMap.entries())) {
             for (const cus of customizations) {
-                // TODO: getTasksToCustomize() will ask all task providers to contribute tasks. Doing this in a loop is bad.
-                const detected = await this.providedTaskConfigurations.getTaskToCustomize(token, cus, rootFolder);
```

Comment:
Comment to reviewers:
- There exists one more calls in the code to `getTaskToCustomize()`
- https://github.com/eclipse-theia/theia/blob/6439a6ddb30ced16fae54d12f9af22f332e8e64e/packages/task/src/browser/task-configurations.ts#L189-L204
- I do not know if that flow also has a related filtering bug or if its working as "intended".
- I'd rather not expand the scope of this PR to other bugs as I am already getting lost in the complexity here...